core/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java

```diff
@@ -1092,6 +1086,7 @@ public void testAcquireIndexCommit() throws Exception {
     public void testSnapshotStore() throws IOException {
         final IndexShard shard = newStartedShard(true);
         indexDoc(shard, "test", "0");
+        shard.updateLocalCheckpointForShard(shard.shardRouting.allocationId().getId(), 0);
```

Comment:
I presume you did this to make sure the global checkpoint advances. I wonder if we should fold into indexDoc? it's sneaky